pkg/rtc/mediatrack.go

```diff
@@ -651,3 +692,50 @@ func (t *MediaTrack) DebugInfo() map[string]interface{} {
 func (t *MediaTrack) Receiver() sfu.TrackReceiver {
 	return t.receiver
 }
+
+func (t *MediaTrack) GetUpConnectionScore() float64 {
+	t.statsLock.Lock()
+	defer t.statsLock.Unlock()
+	return t.connectionStatsUp.Score
+}
+
+func (t *MediaTrack) closeChan() {
+	close(t.done)
+}
+
+func (t *MediaTrack) updateStats() {
+	for {
+		select {
+		case _, ok := <-t.done:
+			if !ok {
+				return
+			}
+		case <-time.After(lastUpdateDelta):
+
+			// take lock on track access buffer
+			t.lock.Lock()
+			var stats buffer.Stats
+			if t.buffer != nil {
+				stats = t.buffer.GetStats()
+			}
+			t.lock.Unlock()
+
+			// lock the stats
+			t.statsLock.Lock()
+			t.connectionStatsUp.Curr.TotalPackets += stats.PacketCount
+			now := time.Now()
+			// calculate score if last update > interval
+			if now.Sub(t.connectionStatsUp.LastUpdated) > lastUpdateDelta {
```

Comment:
Do we want to include on the last second worth of data in each connection quality update? It seems that if we had a larger gap, it'd make it more reliable and less jumpy. wdyt?

Reply:
currently as written, it'll go into this loop every time.

Reply:
will increase the interval to 5 seconds